kroxylicious/src/test/java/io/kroxylicious/proxy/internal/InternalCompletionStageTest.java

```diff
@@ -0,0 +1,308 @@
+        var other = CompletableFuture.<Void> completedFuture(null);
+        return Stream.of(
+                Arguments.of("thenAcceptAsync", (Function<CompletionStage<Void>, CompletionStage<Void>>) (s) -> s.thenAcceptAsync(u -> tallyInvocations())),
+                Arguments.of("thenApplyAsync", (Function<CompletionStage<Void>, CompletionStage<Void>>) (s) -> s.thenApplyAsync(u -> tallyInvocations())),
+                Arguments.of("thenComposeAsync",
+                        (Function<CompletionStage<Void>, CompletionStage<Void>>) (s) -> s.thenComposeAsync(u -> CompletableFuture.completedStage(tallyInvocations()))),
+                Arguments.of("thenRunAsync", (Function<CompletionStage<Void>, CompletionStage<Void>>) (s) -> s.thenRunAsync(() -> tallyInvocations())),
+
+                Arguments.of("handleAsync", (Function<CompletionStage<Void>, CompletionStage<Void>>) (s) -> s.handleAsync((u, t) -> tallyInvocations())),
+                Arguments.of("whenCompleteAsync", (Function<CompletionStage<Void>, CompletionStage<Void>>) (s) -> s.whenCompleteAsync((u, t) -> tallyInvocations())),
+
+                Arguments.of("exceptionallyAsync", (Function<CompletionStage<Void>, CompletionStage<Void>>) (s) -> s.exceptionallyAsync((t) -> tallyInvocations())),
+                Arguments.of("exceptionallyComposeAsync",
+                        (Function<CompletionStage<Void>, CompletionStage<Void>>) (s) -> s
+                                .exceptionallyComposeAsync((t) -> CompletableFuture.completedStage(tallyInvocations()))),
+
+                Arguments.of("acceptEitherAsync", (Function<CompletionStage<Void>, CompletionStage<Void>>) (s) -> s.acceptEitherAsync(other, (u) -> tallyInvocations())),
+                Arguments.of("applyToEitherAsync",
+                        (Function<CompletionStage<Void>, CompletionStage<Void>>) (s) -> s.applyToEitherAsync(other, (u) -> tallyInvocations())),
+
+                Arguments.of("thenAcceptBothAsync",
+                        (Function<CompletionStage<Void>, CompletionStage<Void>>) (s) -> s.thenAcceptBothAsync(other, (u1, u2) -> tallyInvocations())),
+                Arguments.of("thenCombineAsync",
+                        (Function<CompletionStage<Void>, CompletionStage<Void>>) (s) -> s.thenCombineAsync(other, (u1, u2) -> tallyInvocations())),
+
+                Arguments.of("runAfterBothAsync",
+                        (Function<CompletionStage<Void>, CompletionStage<Void>>) (s) -> s.runAfterBothAsync(other, () -> CompletableFuture.completedStage(
+                                tallyInvocations()))),
+                Arguments.of("runAfterEitherAsync",
+                        (Function<CompletionStage<Void>, CompletionStage<Void>>) (s) -> s.runAfterEitherAsync(other, () -> CompletableFuture.completedStage(
+                                tallyInvocations()))));
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource
```

Comment:
nit: This is certainly a question of style. What is the consensus on using a raw method source annotation and letting junit resolve the source vs passing the method name explicitly to the annotation?

I lean towards the latter as its more explicit but appreciate that it doesn't, really add any saftey.

Reply:
I tend to lean towards the implicit - it is just less words to read, and I find it easy to remember that  the method name will match the test's.   The advantage of the explicit suppose is that the Intelij IDE is smart and knows about the fact that it is a method, and makes navigable.

I don't feel strongly.

Reply:
Lets leave it as is then 😁